skimage/color/tests/test_colorlabel.py

```diff
@@ -216,3 +217,22 @@ def test_avg_with_2d_image():
     labels[1:3, 1:3] = 1
     labels[6:9, 6:9] = 2
     assert_no_warnings(label2rgb, labels, image=img, bg_label=0, kind='avg')
+
+
+def test_overlay_saturation():
+    rgb_img = np.random.uniform(size=(10, 10, 3))
+    labels = np.ones((10, 10), dtype=np.int64)
+    labels[5:, 5:] = 2
+    labels[:3, :3] = 0
+    alpha = 0.3
+    rgb = label2rgb(labels, image=rgb_img, alpha=alpha,
+                    bg_label=0, saturation=1)
+    # check that rgb part of input image is preserved, where labels=0
+    assert_array_almost_equal(rgb_img[:3, :3] * (1 - alpha), rgb[:3, :3])
+
+    # now check with saturation=0 that does not match with rgb values
+    rgb = label2rgb(labels, image=rgb_img, alpha=alpha,
+                    bg_label=0, saturation=0)
+
+    with testing.raises(AssertionError):
+        assert_array_almost_equal(rgb_img[:3, :3] * (1 - alpha), rgb[:3, :3])
```

Comment:
```suggestion
    assert_array_almost_equal(rgb_img[:3, :3] * (1 - alpha), rgb[:3, :3])
```

Reply:
The `with testing.raises(AssertionError):` is needed here, because here the we are testing that the arrays are not equal

Reply:
You are right, but this test looks weird then :confused:.
Can't we refactor it in a separate function `test_zero_saturation` asserting that when `saturation` is 0 the returned image is the gray level values where `labels`=0?

Reply:
Sure, this makes sense